src/main/java/racing/domain/Car.java

```diff
@@ -25,16 +25,12 @@ private void validateNameLength(String name) {
         }
     }
 
-    public void moveByRule() {
-        if(moveCheck()) {
+    public void moveByRule(int value) {
+        if(value >= rule.getMoveCondition()) {
```

Comment:
가독성을 위해 해당 문맥을 이동 전략에서 처리하도록 수정하고, 행위를 잘 표현하는 메서드 네이밍을 지어보면 어떨까요?
그럼 랜덤 값 생성 역할을 하는 객체가 해림님이 설계하신 방향과 달라질 수 있지만 고민해 주세요 🙇
```suggestion
        if(rule.isMovable()) {
```

Reply:
moveCondition을 get 할 필요도 없고 코드도 더 깔끔해지는 것 같아요!!